crates/worker/src/partition/state_machine/lifecycle/migrate_journal_table.rs

```diff
@@ -55,13 +55,18 @@ where
                     name: Default::default(),
                 }
                 .into();
+                let mut new_raw_entry = new_entry.encode::<ServiceProtocolV4Codec>();
+                // Journal V1 hasn't append_time, so we set here as timestamp the invocation creation time.
+                // SAFETY: this timestamp is used only for observability, it's fine if it's not agreed among followers.
+                new_raw_entry.header_mut().append_time =
+                    unsafe { self.metadata.timestamps.creation_time() };
```

Comment:
Not a fan of using `unsafe` to denote logical safety issue rather than memory safety risks. I've noted this before to Till and we agreed to phase this out.

No action needed on this particular PR but just using the opportunity to knowledge share.